src/main/java/bio/terra/workspace/service/job/JobBuilder.java

```diff
@@ -56,27 +56,20 @@ public JobBuilder addParameter(String keyName, Object val) {
   /**
    * Submit a job to stairway and return the jobID immediately.
    *
-   * @param duplicateFlightOk indicates how duplicate job IDs should be treated. true indicates this
-   *     should be ignored, false indicates this should be raised as an exception.
    * @return jobID of submitted flight
    */
-  // TODO: this is really a flag indicating "was this ID generated or not". This should be set
-  // at creation of a JobBuilder instead of on submit calls.
-  public String submit(boolean duplicateFlightOk) {
-    return jobServiceRef.submit(flightClass, jobParameterMap, jobId, duplicateFlightOk);
+  public String submit() {
+    return jobServiceRef.submit(flightClass, jobParameterMap, jobId);
   }
 
   /**
    * Submit a job to stairway, wait until it's complete, and return the job result.
    *
    * @param resultClass Class of the job's result
-   * @param duplicateFlightOk indicates how duplicate job IDs should be treated. true indicates this
-   *     should be ignored, false indicates this should be raised as an exception.
    * @return Result of the finished job.
    */
   @Traced
-  public <T> T submitAndWait(Class<T> resultClass, boolean duplicateFlightOk) {
-    return jobServiceRef.submitAndWait(
-        flightClass, jobParameterMap, resultClass, jobId, duplicateFlightOk);
+  public <T> T submitAndWait(Class<T> resultClass) {
```

Comment:
I wonder if this should return JobOrError. Is throwing convenient in some way?
Maybe out of scope for this PR

Reply:
Throwing is mainly a convenience. Services calling `submitAndWait` are almost always executing synchronous flights, so if they get an error instead of a result they would end up throwing themselves.

Reply:
That is fine and we can leave it.

What I was thinking is that we could make a generic result container (JobReport, ErrorReport, and <T> class) with some state accessors and use that more universally rather than gather those parts as the result of a flight and then split them back out again between a throw and a return value. That kind of result container might be a handy common library thing. Hmm...

Anyway, not for this PR!